pyshelf/metadata/yaml_codec.py

```diff
@@ -0,0 +1,23 @@
+import yaml
+
+
+class YamlCodec(object):
+    def serialize(self, metadata):
+        # safe_dump so it doesn't try to represent a python
+        # object in yaml and only serializes native yaml
+        # types.
+        #
+        # default_flow_style so that it doesn't try to stick
+        # inline json for smaller objects.
+        contents = yaml.safe_dump(
+            metadata,
+            encoding="utf-8",
+            indent=4,
+            default_flow_style=False
+        )
+
+        return contents
+
+    def deserialize(self, metadata):
+        metadata = yaml.load(metadata)
+        return metadata
```

Comment:
This raises a question for me. After talking to you and Tyler I am wondering if in cases like this where the line doesn't get super long if we should do one liners - `return yaml.load(metadata)`. Honestly, it doesn't matter but I want to try getting a standard for myself and I am wondering what you think about this.